recordriver/driver.go

```diff
@@ -141,7 +141,15 @@ func (s *stmt) Query(_ []driver.Value) (driver.Rows, error) {
 	sess := s.session
 	sessions[sess].Queries = append(sessions[sess].Queries, s.query)
 	if resp, ok := sessions[sess].responses[s.query]; ok {
-		return resp, nil
+		var copiedResp Response
+		copiedResp.Cols = make([]string, len(resp.Cols))
+		copy(copiedResp.Cols, resp.Cols)
+		copiedResp.Data = make([][]driver.Value, len(resp.Data))
+		for i := range resp.Data {
+			copiedResp.Data[i] = make([]driver.Value, len(resp.Data[i]))
+			copy(copiedResp.Data[i], resp.Data[i])
+		}
+		return &copiedResp, nil
 	}
```

Comment:
I'm pretty sure you can simply write it like this:
```
return &Response{Cols: resp.Cols[:], Data: Cols: resp.Data[:]}
```

Reply:
Indeed, way better now, thanks!